### PR TITLE
Telephone Number Validator

### PR DESCRIPTION
**Pull request by faridafaqiri**
## Pull Request Description
Description:
This pull request enhances the functionality and user experience of the Telephone Number Validator web application. It introduces input validation for various US phone number formats and improves the visual presentation using CSS. Additionally, it adds clear functionality to reset input fields and results.

Changes Made:
Implemented JavaScript validation for US phone numbers.
Styled the application with a phone-themed design using CSS.
Added functionality to clear input fields and results.
Updated README.md with detailed project information, setup instructions, and usage guidelines.

passed the free code camp test and a its also responsive.

telephone num

### CODE REVIEW

**Comment by faridafaqiri** [comment]
@Fikrat2002
hi dear!
I'm waiting for your feedback.

**Comment by Fikrat2002** [comment]
> @Fikrat2002 hi dear! I'm waiting for your feedback.

Hi @far_faqiri 🙋‍♀,

Great job! 💯 You've nailed all the project requirements👏Your feature is complete! Congratulations!!! 🎊🏆

Status: Approved ✅
Everything looks perfect, merge your project 🏆🎊👏

Highlights 💡
🏆 Successfully implemented a great design
✅ Created a professional repository with your project name
✅ Passed FreeCodeCamp requirements for the project 
✅ Used Gitflow correctly by creating a development branch, setting it as default, and making commits on your feature branch
✅ Wrote meaningful commit messages
✅ Set up appropriate linter configurations based on your project's language
✅ No linter errors!
✅ Followed the task and used command-line commits
✅ Submitted a pull request from your feature branch to the development branch and shared the link for review
✅ Provided a clear and detailed description of your work in the PR
✅ Maintained clean code
✅ Created a professional README.md file in your repository
✅ The project is responsive

**Comment by faridafaqiri** [comment]
@ahzamir
Hi dear mentor 
please review my project.

File: script.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    resultsDiv.textContent = `Valid US number: ${userInput}`;
+  } else {
+    resultsDiv.textContent = `Invalid US number: ${userInput}`;
+  }
+});
+
+document.getElementById('clear-btn').addEventListener('click', () => {
+  document.getElementById('user-input').value = '';
+  document.getElementById('results-div').textContent = '';
+});
```
Comment (ahzamir) [inline script.js]: - [OPTIONAL] - Try to add your Event Listeners in while DOM content is loading, it helps to add all of your event listeners to be ready to the action right when your page loads and prevents any side effects below is an example to help you do that:

```javascript
document.addEventListener('DOMContentLoaded', () => {
  //In here add all of your addEventListener functions
  // don't forget to add the constants that you need in your event listeners here as the scope of this one is higher than other codes
  // As an Example I added only one Event Listener example here, you add your own:
  
  const clearBtn = document.getElementById('clear-btn');
  const resultsDiv = document.getElementById('results-div');
  
  clearBtn.addEventListener('click', () => {
    resultsDiv.innerHTML = '';
  });
})
```

**Review by ahzamir — CHANGES_REQUESTED** [review]
### Hi @faridafaqiri 🙋‍♀️,

> Excellent work! 💯 You have done a great job implementing all the project requirements. 🥳 Your feature is complete! Congratulations!!!🎊

- Well done you have finished lots of requirements but still your code misses some part of the requirements and you need to work on it please check the comments below for more details:

- **[REQUIRED]** Share your project on this [GROUP](https://www.facebook.com/groups/kick-startdev), add a simple Proud Moment text and a screenshot of your project then besides that add your project link at the end of the text as a GitHub Source code reference and post.

- Share your Project on your LinkedIn account and add a proud moment text with a screenshot.

- Well Done great design Implemented. 🎯🎉👍
image

# Status: Required Changes 🔁
Make changes to your code and commit and request for a re-review.

AlmostThereReachGIF
b64e13140efa)

 ## Highlights 🌞

🎉 Great Design Implemented 
✔️ Created a professional Repo (You created a repository with your project name)
✔️ Correct use of Gitflow (You created the development branch and made it your default branch, then created and committed your work on your feature branch)
✔️ Commit messages are meaningful
✔️ Setup the linters correctly (You added the correct linters config into your project according to the language you used on your code)
✔️ No linter errors (All linters checks have been passed 🟢)
✔️ You have followed the Telephone Number Validator project task and committed using the command line
✔️ Created the pull request (From your feature branch to the development branch you created a pull request and submitted the pull request link for review)
✔️ Good PR description (You added a meaningful description to your PR about your work and the features you added on this pull request)
❌Your code is clean
✔️Your repo has a professional README.md file
✔️ Your partner has reviewed your repo
❌Your design is responsive

**Cheers 🥂 and Happy coding!!! 👯 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me @ahzamir in your question so I can receive the notification.**

File: styles.css
Comment (ahzamir) [inline styles.css:1]: - Although your design is looking very great but still your design is not responsive:
image

**Review by ahzamir — CHANGES_REQUESTED** [review]

**Comment by Fikrat2002** [comment]
> 
> 
> 
> 
> ### Hi @faridafaqiri 🙋‍♀️,
> 
> 
> 
> > Excellent work! 💯 You have done a great job implementing all the project requirements. 🥳 Your feature is complete! Congratulations!!!🎊
> 
> 
> 
> - Well done you have finished lots of requirements but still your code misses some part of the requirements and you need to work on it please check the comments below for more details:
> 
> 
> 
> - **[REQUIRED]** Share your project on this [GROUP](https://www.facebook.com/groups/kick-startdev), add a simple Proud Moment text and a screenshot of your project then besides that add your project link at the end of the text as a GitHub Source code reference and post.
> 
> 
> 
> - Share your Project on your LinkedIn account and add a proud moment text with a screenshot.
> 
> 
> 
> - Well Done great design Implemented. 🎯🎉👍
> 
> image
> 
> 
> 
> # Status: Required Changes 🔁
> 
> Make changes to your code and commit and request for a re-review.
> 
> 
> 
> AlmostThereReachGIF
> 
> b64e13140efa)
> 
> 
> 
>  ## Highlights 🌞
> 
> 
> 
> 🎉 Great Design Implemented 
> 
> ✔️ Created a professional Repo (You created a repository with your project name)
> 
> ✔️ Correct use of Gitflow (You created the development branch and made it your default branch, then created and committed your work on your feature branch)
> 
> ✔️ Commit messages are meaningful
> 
> ✔️ Setup the linters correctly (You added the correct linters config into your project according to the language you used on your code)
> 
> ✔️ No linter errors (All linters checks have been passed 🟢)
> 
> ✔️ You have followed the Telephone Number Validator project task and committed using the command line
> 
> ✔️ Created the pull request (From your feature branch to the development branch you created a pull request and submitted the pull request link for review)
> 
> ✔️ Good PR description (You added a meaningful description to your PR about your work and the features you added on this pull request)
> 
> ❌Your code is clean
> 
> ✔️Your repo has a professional README.md file
> 
> ✔️ Your partner has reviewed your repo
> 
> ❌Your design is responsive
> 
> 
> 
> **Cheers 🥂 and Happy coding!!! 👯 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me @ahzamir in your question so I can receive the notification.**
> 
> 
> 
> 

Hello dear mentor, I just want to mention that I don't understand what the problem exactly is but when I checked Farida's project her design was responsive

**Comment by ahzamir** [comment]
@Fikrat2002 

- You have to check for responsiveness in the inspect:
  - See what result I get from inspect the mobile version:
  
image

**Comment by ahzamir** [comment]
- The bottom of the page does not have background

**Comment by faridafaqiri** [comment]
@ahzamir 
which part of the code is not clean?
I checked but all of my is clean.

**Comment by faridafaqiri** [comment]
@Fikrat2002  
please give me feedback again .

**Comment by faridafaqiri** [comment]
@ahzamir 
I have made responsive my project and clean the code also made changes in JavaScript code.

**Comment by Fikrat2002** [comment]
> @Fikrat2002 
> 
> 
> 
> - You have to check for responsiveness in the inspect:
> 
>   - See what result I get from inspect the mobile version:
> 
>   
> 
> image
> 
> 

You were right🥲

**Comment by Fikrat2002** [comment]
> > @Fikrat2002 
> > 
> > 
> > 
> > - You have to check for responsiveness in the inspect:
> > 
> >   - See what result I get from inspect the mobile version:
> > 
> >   
> > 
> > image
> > 
> > 
> 
> You were right🥲

@faridafaqiri 
You have applied the feedbacks and I see no further issues

**Comment by faridafaqiri** [comment]
@ahzamir hi 
teacher please give me feedback.

**Review by ahzamir — APPROVED** [review]
### Hi @faridafaqiri 🙋‍♀️,

> Excellent work! 💯 You have done a great job implementing all the project requirements. 🥳 Your feature is complete! Congratulations!!!🎊

- **[REQUIRED]** Share your project on this [GROUP](https://www.facebook.com/groups/kick-startdev), add a simple Proud Moment text and a screenshot of your project then besides that add your project link at the end of the text as a GitHub Source code reference and post.

- Share your Project on your LinkedIn account and add a proud moment text with a screenshot.

- Well Done great design Implemented. 🎯🎉👍
image

# Status: Approved ✅
There is nothing to say other than to merge your branch. 🎉🎯✅👍😊🥳

WinkThumbsUpGIF

 ## Highlights 🌞

🎉 Great Design Implemented 
✔️ Created a professional Repo (You created a repository with your project name)
✔️ Correct use of Gitflow (You created the development branch and made it your default branch, then created and committed your work on your feature branch)
✔️ Commit messages are meaningful
✔️ Setup the linters correctly (You added the correct linters config into your project according to the language you used on your code)
✔️ No linter errors (All linters checks have been passed 🟢)
✔️ You have followed the Telephone Number Validator project task and committed using the command line
✔️ Created the pull request (From your feature branch to the development branch you created a pull request and submitted the pull request link for review)
✔️ Good PR description (You added a meaningful description to your PR about your work and the features you added on this pull request)
✔️Your code is clean
✔️Your repo has a professional README.md file
✔️ Your partner has reviewed your repo
✔️Your design is responsive

**Cheers 🥂 and Happy coding!!! 👯 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me @ahzamir in your question so I can receive the notification.**

**Pull request merged**